10up/classifai
# Amazon Polly as a provider for the text-to-speech feature.

### Description of the Change
PR Adds Amazon Polly as a provider for the Text-to-speech feature.

Closes #728 

@jeffpaul @dkotter Amazon Polly provides additional options such as [Newscaster Speaking Style](https://docs.aws.amazon.com/polly/latest/dg/ntts-speakingstyles.html) and [SSML](https://docs.aws.amazon.com/polly/latest/dg/ssml.html), which offers features like including breathing sounds and emphasizing specific words or phrases. I haven't implemented it in this PR, but we can consider integrating it in the future if there are specific client requirements around these features.

@dkotter, I haven't added E2E tests for this because I haven't figured out yet how to mock the API, given that we are using the AWS PHP SDK here. Please let me know if you have any ideas on this.

### How to test the Change
1. Go to Tools > ClassifAI > Language Processing > Text to Speech.
1. Select the "Amazon Polly" option as the provider.
1. Add AWS credentials and save settings.
1. Create/Edit a post and ensure that the Text-to-Speech feature is working as expected.

### Changelog Entry
> Added - Amazon Polly as a provider for the text-to-speech feature.


### Credits
Props @jeffpaul @iamdharmesh 


### Checklist:
